Source/ACE.Server/WorldObjects/Managers/EnchantmentManager.cs

```diff
@@ -721,7 +729,7 @@ public virtual float GetVitalMod_Multiplier(CreatureVital vital)
         /// </summary>
         public virtual float GetXPBonus()
         {
-            var enchantments = GetEnchantments(SpellCategory.TrinketXPRaising);
+            var enchantments = GetEnchantments_TopLayer(SpellCategory.TrinketXPRaising);
```

Comment:
if this function doesn't already exist, then it should probably just sort by descending StatModVal, and return the first one

Reply:
foreach (var enchantment in enchantments.OrderByDescending(i => i.StatModValue).Take(1))